File: frontend/src/lib/types/crosslink.ts
export interface CrossLinkPeer {
	host: string;
	target_id: string | null;
	target_value: string;
}

export interface CrossLink {
	kind: string;
	label: string;
	value: string;
	query: string;
	peers: CrossLinkPeer[];
	targets: string[];
	hosts: number;
}

export function linkedTargets(links: CrossLink[]): string[] {
	return [...new Set(links.flatMap((l) => l.targets))].filter(Boolean).sort();
}

File: shared/models/crosslink.py
import uuid

from pydantic import BaseModel, Field


class CrossLinkPeer(BaseModel):
    host: str
    target_id: uuid.UUID | None = None
    target_value: str = ""


class CrossLink(BaseModel):
    kind: str
    label: str
    value: str
    query: str
    peers: list[CrossLinkPeer] = Field(default_factory=list)
    targets: list[str] = Field(default_factory=list)
    hosts: int = 0
